signature-aggregator/main/main.go

```diff
@@ -151,11 +155,27 @@ func main() {
 	healthcheck.HandleHealthCheckRequest(networkHealthcheckFunc)
 
 	logger.Info("Initialization complete")
-	err = http.ListenAndServe(fmt.Sprintf(":%d", cfg.APIPort), nil)
-	if errors.Is(err, http.ErrServerClosed) {
-		logger.Info("Server closed")
-	} else if err != nil {
-		logger.Fatal("Server error", zap.Error(err))
+	errGroup.Go(func() error {
+		httpServer := &http.Server{
+			Addr: fmt.Sprintf(":%d", cfg.APIPort),
+		}
+		// Handle Graceful shutshown
+		go func() {
+			<-ctx.Done()
+			if err := httpServer.Shutdown(ctx); err != nil {
+				logger.Error("Failed to shutdown healthcheck server", zap.Error(err))
```

Comment:
This isn't just the healthcheck server, right? It's the whole server including the main `aggregate-signatures` endpoint. 

Otherwise these changes look better